cwl/src/main/scala/cwl/WorkflowStepInputExpression.scala

```diff
@@ -47,7 +82,8 @@ final case class WorkflowStepInputExpression(input: WorkflowStepInput,
        * input parameters is undefined and the result of evaluating valueFrom on a parameter must not be visible to
        * evaluation of valueFrom on other parameters.
        */
-      case (Some(StringOrExpression.Expression(expression)), Some(sources)) =>
+      case (Some(StringOrExpression.Expression(expression)), Some(sources), _) =>
```

Comment:
What about `Some(StringOrExpression.String(...)), Some(sources))`? Thibault actually has CWL like that in the comment of his open PR. 🙂

Reply:
Well I should put this case in, but it looks like I will only be using the string value and ignoring those sources:

>If valueFrom is a constant string value, use this as the value for this input parameter.

Reply:
So it seems like we should throw an error in this situation.  If someone specifies sources and has a string value in their expression, it's almost certainly a bug in their semantics (i.e. expecting to use the sources).

Reply:
I will write an informative error message in this case